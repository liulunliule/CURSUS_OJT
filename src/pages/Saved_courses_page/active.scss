.active {
    margin-left: 0;

    .saved_courses_cta {
        margin-right: 21px;

        &-title {
            column-gap: 187px;
        }
    }

    .saved_courses_line {
        width: 350px;
    }
    .saved_courses_cta-desc-btn {
        margin-top: 30px;
        display: flex;
        align-items: center;
        color: #fff;
        padding: 8px 91px;
        background-color: #ed2a26;
        border: none;
        outline: none;
        font-size: 14px;
        border-radius: 3px;
    }

    .saved_courses_media-item {
        width: 1092px;
    }

    .saved_courses_media-thumbnail {
        .thumbnail {
            width: 428px;
            height: 241px;
            object-fit: cover;
        }

        .saved_courses_media-overlay {
            width: 428px;
            height: 241px;
        }
    }
    .saved_courses_media-content-views {
        .dots {
            margin-left: 469px;
        }
    }
}
