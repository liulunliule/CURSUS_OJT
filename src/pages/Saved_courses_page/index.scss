.saved_courses {
    margin-top: 60px;
    margin-left: 240px;
    padding: 30px 30px 50px 30px;
    background-color: #f7f7f7;
    transition: margin-left 0.5s ease;

    &_inner {
        display: flex;
    }

    &_cta {
        margin-right: 21px;

        &-title {
            display: flex;
            column-gap: 127px;

            &-left {
                color: #333;
                font-size: 14px;
                font-weight: 500;
                line-height: 19.999px;
                white-space: nowrap;
            }

            &-right {
                color: #000000de;
                font-size: 14px;
                line-height: 19.999px;
                white-space: nowrap;
            }
        }

        &-desc {
            margin-top: 25px;

            .saved_courses_quantity {
                color: #686f7a;
                font-size: 14px;
                line-height: 19.999px;
            }
            .saved_courses_cta-desc-btn {
                margin-top: 30px;
                display: flex;
                align-items: center;
                color: #fff;
                padding: 8px 60px;
                background-color: #ed2a26;
                border: none;
                outline: none;
                font-size: 14px;
                border-radius: 3px;
                white-space: nowrap;
            }
            .saved_courses_cta-icon {
                color: #fff;
                margin-right: 7px;
            }
        }
    }

    &_line {
        margin-top: 5px;
        width: 290px;
        height: 1px;
        background: #e5e5e5;
    }

    &_media-title {
        margin-top: -3px;
        color: #000000de;
        font-size: 16px;
        font-weight: 500;
        margin-bottom: 20px;
    }

    &_media-item {
        position: relative;
        display: flex;
        background-color: #fff;
        padding: 10px;
        width: 912px;
        border: 0.8px solid #efefef;
        border-radius: 3px;
        margin-bottom: 31px;
        transition: transform 0.5s ease;
        transition: width 0.5s ease, height 0.5s ease;

        &:hover {
            transform: translateY(1px);
            cursor: pointer;
        }
    }

    &_media-thumbnail {
        margin-right: 20px;
        position: relative;

        .thumbnail {
            width: 356px;
            height: 200px;
            object-fit: cover;
            transition: width 0.5s ease, height 0.5s ease;
        }

        .saved_courses_media-overlay {
            width: 356px;
            height: 200px;
            background-image: linear-gradient(
                transparent,
                rgba(51, 51, 51, 0.3) 90%
            );
            position: absolute;
            top: 0;
            left: 0;
            right: 0;
            transition: width 0.5s ease, height 0.5s ease;
        }

        .saved_courses_media-star {
            display: inline-flex;
            align-items: center;
            gap: 4px;
            position: absolute;
            top: 10px;
            left: 10px;
            color: #fff;
            background: #fdcc0d;
            padding: 3px 15px 3px 10px;
            border-radius: 3px;
            font-size: 14px;
            font-weight: 500;
        }

        .saved_courses_media-play {
            border: 0;
            width: 50px;
            height: 50px;
            float: left;
            text-align: center;
            border-radius: 100%;
            padding: 15px 13px;
            font-size: 20px;
            color: #fff;
            position: absolute;
            top: 40%;
            right: 40%;
            opacity: 0;
            background: rgba(51, 51, 51, 0.8);
            transition: opacity 0.5s ease;

            &::before {
                position: absolute;
                content: url(../../assets/img/Saved_Courses/play-7-2.png);
                top: 25%;
                right: 32%;
            }
        }

        .saved_courses_media-seller {
            border-radius: 3px;
            font-size: 10px;
            position: absolute;
            right: 0;
            top: 10px;
            padding: 2px 10px;
            background: #fa8305;
            border: none;
            color: #fff;
            font-weight: 600;
            z-index: 2;
            border-bottom-right-radius: 0;
            border-top-right-radius: 0;
            box-shadow: 0 0 1px 1px rgba(20, 23, 28, 0.1),
                0 3px 1px 0 rgba(20, 23, 28, 0.1);

            &::after {
                background: inherit;
                content: "";
                height: 12px;
                position: absolute;
                top: 3px;
                transform: rotate(45deg);
                width: 12px;
                z-index: 1;
                display: block;
                left: -5px;
                border-bottom-left-radius: 2px;
                border-top-left-radius: 3px;
                border-bottom-right-radius: 3px;
            }
        }

        .saved_courses_media-timer {
            bottom: 0;
            position: absolute;
            margin-bottom: 10px;
            right: 0;
            margin-right: 10px;
            color: #fff;
            background: rgba(51, 51, 51, 0.8);
            display: inline-block;
            padding: 4px 10px;
            border-radius: 3px;
            font-size: 12px;
            font-weight: 500;
        }
    }
    &_media-content {
        &-views {
            margin-top: 7px;
            display: flex;
            align-items: center;

            .view-left,
            .view-right {
                font-size: 12px;
                color: #686f7a;
                white-space: nowrap;
            }

            .view-left::after {
                content: "â€¢";
                margin: 0px 4px;
            }
            .dots {
                padding: 0 2px;
                margin-left: 362px;
                font-size: 25px;
                line-height: 19.999px;
                color: #afafaf;
                transition: margin-left 0.5s ease;

                &:hover {
                    color: #333;
                }
                &:hover .dots_popup {
                    visibility: visible;
                }
            }
        }
        &-title {
            margin-top: 11px;
            font-size: 16px;
            font-weight: 600;
            color: #333;
            line-height: 24px;
        }

        &-desc {
            margin-top: 7px;
            font-size: 13px;
            font-weight: 400;
            color: #686f7a;
            padding-top: 2px;
        }

        &-buy {
            display: flex;
            margin-top: 18px;
            align-items: center;
            justify-content: space-between;

            .author {
                color: #686f7a;
                font-size: 14px;

                a {
                    color: #333;
                    font-weight: 500;
                }
            }

            .deal {
                display: flex;
                align-items: center;
                .cart {
                    margin-right: 10px;
                    border: none;
                    outline: none;
                    background-color: #fff;
                    opacity: 0;
                    transition: color 0.5s ease;
                    &:hover {
                        color: #ed2a26;
                    }
                }
                .cost {
                    font-size: 18px;
                    font-weight: 500;
                    color: #333;
                }
            }
        }
    }
}

.saved_courses_media-item:hover .cart {
    opacity: 1;
}

.saved_courses_media-thumbnail:hover .saved_courses_media-play {
    opacity: 1;
}

.dots_popup {
    display: flex;
    align-items: center;
    padding: 10px;
    column-gap: 10px;
    position: absolute;
    top: 40px;
    right: 11px;
    width: 158px;
    height: 41px;
    background-color: #fff;
    border: 1px solid #e5e5e5;
    box-shadow: 3px 7px 5px -1px rgba(0, 0, 0, 0.2);
    font-size: 14px;
    color: #686f7a;
    visibility: hidden;

    &:hover {
        background-color: #ffecec;
        color: #333;
    }

    .closeIcon {
        width: 11px;
        color: #686f7a;

        &:hover {
            color: #333;
        }
    }
}

// .dots:hover .dots_popup {
//     opacity: 1;
// }
