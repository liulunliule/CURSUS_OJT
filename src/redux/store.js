import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
