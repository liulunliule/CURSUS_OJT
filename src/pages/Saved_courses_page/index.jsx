import React, { useEffect } from "react";
import "./index.scss";
import "./active.scss";
import {
    CloseOutlined,
    DeleteOutlined,
    ShoppingCartOutlined,
    StarOutlined,
} from "@ant-design/icons";
import { Link, useNavigate } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import {
    fetchCourse,
    removeSavedCourse,
} from "../../redux/features/courseSlice";
import { toast } from "react-toastify";
import axios from "axios";

function Saved_courses_page() {
    const dispatch = useDispatch();
    const account = useSelector((state) => state.user.account);
    const savedCourses = useSelector((state) => state.course.courses);
    const isShowAll = useSelector((state) => state.savedCourse.isShowAll);

    useEffect(() => {
        if (account.id) {
            dispatch(fetchCourse(account.id));
        }
    }, [dispatch, account.id]);

    const handleRemoveSavedCourse = async (courseId) => {
        // dispatch(removeSavedCourse(courseId));
        console.log("Id: ", courseId);
        await axios.delete(
            `https://6696231a0312447373c1386e.mockapi.io/savedCourse/${courseId}`
        );
        const listAfterDelete = savedCourses.filter(
            (course) => course.id !== courseId
        );
        return listAfterDelete;
    };

    return (
        <div className={`saved_courses ${isShowAll ? "active" : ""}`}>
            <div className="saved_courses_inner">
                <div className="saved_courses_cta">
                    <div className="saved_courses_cta-title">
                        <p className="saved_courses_cta-title-left">
                            Saved Courses
                        </p>
                        <p className="saved_courses_cta-title-right">
                            Remove All
                        </p>
                    </div>
                    <div className="saved_courses_line"></div>
                    <div className="saved_courses_cta-desc">
                        <div className="saved_courses_quantity">
                            {savedCourses.length} Courses
                        </div>
                        <button className="saved_courses_cta-desc-btn">
                            <div className="saved_courses_cta-icon">
                                <DeleteOutlined />
                            </div>
                            Remove Saved Courses
                        </button>
                    </div>
                </div>
                <div className="saved_courses_media">
                    <p className="saved_courses_media-title">Saved Courses</p>
                    <div className="saved_courses_media-list">
                        {savedCourses.map((savedCourse) => (
                            <div
                                className="saved_courses_media-item"
                                key={savedCourse.id}
                            >
                                <Link
                                    to={`/course_detail/${savedCourse.id}/course_detail_about`}
                                >
                                    <div className="saved_courses_media-thumbnail">
                                        <img
                                            src={savedCourse.video}
                                            alt=""
                                            className="thumbnail"
                                        />
                                        <div className="saved_courses_media-overlay"></div>
                                        <div className="saved_courses_media-star">
                                            <div className="star">
                                                <StarOutlined />
                                            </div>
                                            {savedCourse.ratting}
                                        </div>
                                        <div className="saved_courses_media-play"></div>
                                        <div className="saved_courses_media-seller">
                                            {savedCourse.level}
                                        </div>
                                        <div className="saved_courses_media-timer">
                                            {savedCourse.time} hours
                                        </div>
                                    </div>
                                </Link>

                                <div className="saved_courses_media-content">
                                    <div className="saved_courses_media-content-views">
                                        <div className="view-left">
                                            {savedCourse.views}
                                        </div>
                                        <div className="view-right">
                                            {savedCourse.date} days ago
                                        </div>
                                        <div className="dots">
                                            â‹®
                                            <div
                                                className="dots_popup"
                                                onClick={() =>
                                                    handleRemoveSavedCourse(
                                                        savedCourse.id
                                                    )
                                                }
                                            >
                                                <CloseOutlined className="closeIcon" />
                                                Remove
                                            </div>
                                        </div>
                                    </div>
                                    <Link
                                        to={`/course_detail/${savedCourse.id}/course_detail_about`}
                                    >
                                        <div className="saved_courses_media-content-title">
                                            {savedCourse.titilecourse}
                                        </div>
                                    </Link>
                                    <div className="saved_courses_media-content-desc">
                                        {savedCourse.typecourse}
                                    </div>

                                    <div className="saved_courses_media-content-buy">
                                        <p className="author">
                                            By <Link>{savedCourse.author}</Link>{" "}
                                        </p>
                                        <div className="deal">
                                            <Link
                                                to={`/course_detail/${savedCourse.id}/course_detail_about`}
                                            >
                                                <button className="cart">
                                                    <ShoppingCartOutlined />
                                                </button>
                                            </Link>
                                            <div className="cost">
                                                ${savedCourse.price}
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    );
}

export default Saved_courses_page;
