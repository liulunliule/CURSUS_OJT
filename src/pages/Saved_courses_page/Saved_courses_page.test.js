import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Saved_courses_page from '.';

test('test Saved courses page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Saved_courses_page/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
