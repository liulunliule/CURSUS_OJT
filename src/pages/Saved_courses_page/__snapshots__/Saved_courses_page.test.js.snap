// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Saved courses page 1`] = `
<div
  className="saved_courses "
>
  <div
    className="saved_courses_inner"
  >
    <div
      className="saved_courses_cta"
    >
      <div
        className="saved_courses_cta-title"
      >
        <p
          className="saved_courses_cta-title-left"
        >
          Saved Courses
        </p>
        <p
          className="saved_courses_cta-title-right"
        >
          Remove All
        </p>
      </div>
      <div
        className="saved_courses_line"
      />
      <div
        className="saved_courses_cta-desc"
      >
        <div
          className="saved_courses_quantity"
        >
          0
           Courses
        </div>
        <button
          className="saved_courses_cta-desc-btn"
        >
          <div
            className="saved_courses_cta-icon"
          >
            <span
              aria-label="delete"
              className="anticon anticon-delete"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="delete"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                />
              </svg>
            </span>
          </div>
          Remove Saved Courses
        </button>
      </div>
    </div>
    <div
      className="saved_courses_media"
    >
      <p
        className="saved_courses_media-title"
      >
        Saved Courses
      </p>
      <div
        className="saved_courses_media-list"
      />
    </div>
  </div>
</div>
`;
